pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +54,91 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
-	if cmdutil.GetFlagBool(cmd, "short") {
-		v = version.Get().GitVersion
+	vo := VersionObj{nil, nil}
+
+	clientVersion := version.Get()
+	vo.ClientVersion = &clientVersion
+
+	serverVersion, serverErr := retrieveServerVersion(f)
```

Comment:
put the fetch inside the `if` block... we don't want to hit the server if they've told us they only want the client version